### ISSUE TITLE
Explode parameter is not correctly converted from OpenAPI 3.0 to 3.1 documents

### ISSUE THREAD

**Issue by czechboy0** [apple/swift-openapi-generator#377]
The exploded query parameter style is not working in current version (0.3.4). It used to work properly in previous version (0.1.11)

example openapi.yaml
```
openapi: '3.0.3'
info:
  description:
  version: 2023.8.26
  title: example
  
servers:
  - url: https://example
  
paths:
  /example:
    get:
      operationId: example
      parameters:
      - name: LocationId
        required: false
        in: query
        style: form
        explode: false
        schema:
          type: array
          items:
            type: string
```

It now yields:
`https://example/example?LocationId=1&LocationId=2`

But it is expected to be:
`https://example/example?LocationId=1,2`

_Originally posted by @luomein in https://github.com/apple/swift-openapi-generator/issues/258#issuecomment-1808507532_

**Comment by czechboy0** [comment]
Thanks for reporting this @luomein, this indeed looks like a bug.

I've identified that it only happens for documents using OpenAPI 3.0, as we then convert them to 3.1 in memory using OpenAPIKit's `to31` method under the hood, however the one for parameter context seems _not_ to be bringing over the `explode` parameter: https://github.com/mattpolzin/OpenAPIKit/blob/f93e9bd8ec7b9d58f6a86e303fd5f71c7088f9ae/Sources/OpenAPIKitCompat/Compat30To31.swift#L171 @mattpolzin could you take a look?

In the meantime, you can resolve this by using `openapi: 3.1.0` at the top of your document, @luomein.

**Comment by luomein** [comment]
Thanks for the quick response. Confirmed it works after updating openapi: '3.1.0' in my openapi.yaml

**Comment by czechboy0** [comment]
@luomein it was also fixed in OpenAPIKit 3.1.0, if you update the dependency, can you check it then also works with a document marked with the OpenAPI version 3.0.x?

https://github.com/mattpolzin/OpenAPIKit/releases/tag/3.1.0

**Comment by luomein** [comment]
> @luomein it was also fixed in OpenAPIKit 3.1.0, if you update the dependency, can you check it then also works with a document marked with the OpenAPI version 3.0.x?
> 
> https://github.com/mattpolzin/OpenAPIKit/releases/tag/3.1.0

Yes, confirmed. It works with OpenAPIKit 3.1.0 and openapi: '3.0.3' in openapi.yaml, explode: false
